src/api/classes/Methods.ts

```diff
@@ -0,0 +1,67 @@
+import { Integration } from '../../integration';
+import { HTTPMethodValue } from '../props';
+
+export interface IMethodIntegrations {
+  get?: Integration;
+  post?: Integration;
+  put?: Integration;
+  delete?: Integration;
+  options?: Integration;
+  head?: Integration;
+  patch?: Integration;
+  trace?: Integration;
+};
+
+/**
+ * Methods with integrations.
+ *
+ * Has to be a class because of JSII limitations.
+ *
+ * Because of JSII5000 all getters are renamed to non-getXxx named methods.
+ */
+export class Methods {
+  private readonly methods: IMethodIntegrations;
+
+  public constructor(methods: IMethodIntegrations) {
+    this.methods = methods;
```

Comment:
Maybe add a validation check that at least one of the integrations has been set.

Something like `Object.keys.lenght > 0`.